extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,242 @@
+	result += c;
+}
+
+static string ReplaceCharref(string &str) {
+	RE2::Options options;
+	options.set_log_errors(false); // disable error logging for performance
+	options.set_case_sensitive(false);
+	const RE2 charref(R"(&(#[0-9]+;?|#[xX][0-9a-fA-F]+;?|[^\t\n\f <&#;]{1,32};?))", options);
+
+	string match;
+	string result {""};
+	int start_pos = 0;
+	duckdb_re2::StringPiece input(str); // wrap a StringPiece around it
+	while (RE2::FindAndConsume(&input, charref, &match)) {
+		// include the input value until the match
+		size_t match_pos = str.find(match, start_pos) - 1; // include the & ampersand charq
+		result.append(str, start_pos, match_pos - start_pos);
+
+		// the position after the end of the current match
+		start_pos = (match_pos + 1) + match.length();
+
+		int num;
+		if (match[0] == '#') {
+			try {
+				// numeric character reference
+				if (match[1] == 'x' || match[1] == 'X') {
+					num = stoi(match.substr(2), nullptr, 16);
+				} else {
+					num = stoi(match.substr(1));
+				}
+			} catch (const std::out_of_range &e) {
+				result += "\uFFFD";
+				continue;
+			}
+
+			auto invalid_charrefs = ReturnInvalidCharrefs();
+			auto invalid_codepoints = ReturnInvalidCodepoints();
+			if (invalid_charrefs->find(num) != invalid_charrefs->end()) {
+				auto ch = invalid_charrefs->at(num);
+				if (Utf8Proc::Analyze(ch.c_str(), ch.length()) == UnicodeType::INVALID) {
+					auto blob = string_t(invalid_charrefs->at(num));
+					auto str_blob = Blob::ToString(blob);
+					result += str_blob;
+				} else {
+					result += ch;
+				}
+
+			} else if ((0xD800 <= num && num <= 0xDFFF) || 0x10FFFF < num) {
+				result += "\uFFFD";
+			} else if (invalid_codepoints->find(num) != invalid_codepoints->end()) {
+				// skip this character
+				continue;
+			} else {
+				result += static_cast<char>(num);
+			}
+		} else {
+			// named charref
+			auto html5 = ReturnHTML5NameCharrefs();
+			auto it = html5->find(match);
+			if (it != html5->end()) {
+				add_blob(it->second, result);
+				continue;
+			}
+
+			for (int x = match.length() - 1; x >= 1; --x) {
+				string substr = match.substr(0, x);
+				it = html5->find(substr);
+				if (it != html5->end()) {
+					add_blob(it->second, result);
+					start_pos = x + 1;
+					break;
+				}
+			}
+			if (it == html5->end()) {
+				result.append('&' + match);
+			}
+		}
+	}
+	result += str.substr(start_pos);
+	return result;
+}
+
+void INetFunctions::Unescape(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	UnaryExecutor::Execute<string_t, string_t>(args.data[0], result, args.size(), [&](string_t input_st) {
+		auto str = input_st.GetString();
+		if (str.find('&') == str.npos) {
+			return StringVector::AddString(result, str);
+		}
+		string unescaped = ReplaceCharref(str);
+		return StringVector::AddString(result, unescaped);
+	});
+}
+
+ScalarFunctionSet InetExtension::GetEscapeFunctionSet() {
+	ScalarFunctionSet funcs("escape");
```

Comment:
Can we call these functions `html_escape`/`html_unescape` instead? `escape` is a rather generic/non-descriptive name. In Python they are namespaced which allows these names (i.e. `html.escape`)